### PR TITLE
Issue #204

### LINKED ISSUE #204: MethodValidator.java:152-156: We need to find a...
Puzzle `194-8b58eead` in `src/main/java/com/jcabi/aspects/aj/MethodValidator.java:152-156` has to be resolved: We need to find a way to disable the constraint rule OverridingMethodMustNotAlterParameterConstraints instead of catching the exception. See https://hibernate.atlassian.net/browse/HV-1044 After that don't forget to enable test JSR303Test#overridesMessage()... The puzzle was created by Dmitry Zaytsev on 8-Jan-2016.

If you have any technical questions, don't ask me, submit new tickets instead. The task will be "done" when the problem is fixed and the text of the puzzle is removed from the source code.

### PR DESCRIPTION
**Pull request by gmandnepr**
Related issue: https://github.com/jcabi/jcabi-aspects/issues/204

hibernate-validator library upgraded to enable allow_parameter_constraint_override.

Current version that supports this feature is alpha. Should this task be frozen until 5.3 released or new puzzle created for this?

### CODE REVIEW

**Comment by dmarkov** [comment]
@gmandnepr Many thanks, I will find someone to review it before we merge

**Comment by dmarkov** [comment]
@mkordas please review this PR

**Comment by mkordas** [comment]
@gmandnepr I'm on it

File: pom.xml [hunk 1]
```diff
@@ -88,7 +88,7 @@
         <dependency>
             <groupId>org.hibernate</groupId>
             <artifactId>hibernate-validator</artifactId>
-            <version>5.2.2.Final</version>
+            <version>5.3.0.Alpha1</version>
```
Comment (mkordas) [inline pom.xml]: @gmandnepr do you think it is fine to use alpha version? Maybe we should use latest stable one?

File: src/main/resources/META-INF/validation.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<validation-config
+        xmlns="http://jboss.org/xml/ns/javax/validation/configuration"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://jboss.org/xml/ns/javax/validation/configuration validation-configuration-1.0.xsd">
+    <default-provider>org.hibernate.validator.HibernateValidator</default-provider>
+    <property name="hibernate.validator.allow_parameter_constraint_override">true</property>
+</validation-config>
```
Comment (mkordas) [inline src/main/resources/META-INF/validation.xml]: @gmandnepr missing new line at the end of file

**Comment by mkordas** [comment]
@gmandnepr please put link to issue in PR description

File: pom.xml (hunk 1 above)
Comment (mkordas) [inline pom.xml]: @gmandnepr sorry, I've just read description of PR. I think it's fine to use alpha provided all tests are passing

**Comment by mkordas** [comment]
@gmandnepr please see 3 remarks from my side above

**Comment by gmandnepr** [comment]
@mkordas done. 
Should I remove a line from PUZZLES.md with this issue?
Should I add a new issue to puzzles to update the version one it released from alpha?

**Comment by mkordas** [comment]
@gmandnepr 
> Should I remove a line from PUZZLES.md with this issue?

No, @dmarkov will do it
> Should I add a new issue to puzzles to update the version one it released from alpha?

I think VersionEye will show us update and someone will pick this up anyway.

**Comment by mkordas** [comment]
@gmandnepr please do not use word "fixes" in PR description as GitHub will automatically close the issue after merge, while only @dmarkov should close puzzle issues

**Comment by mkordas** [comment]
@gmandnepr I meant PR description, not PR title. GitHub will still close the issue after merge.

File: src/main/java/com/jcabi/aspects/aj/MethodValidator.java [hunk 3]
```diff
@@ -149,28 +148,17 @@ public void after(final JoinPoint point, final Object result) {
...
-            if (!ex.getMessage().startsWith("HV000151")) {
-                throw new ConstraintDeclarationException(ex);
-            }
-        }
+        this.checkForViolations(this.validator
+            .forExecutables()
+            .validateParameters(
+                object,
+                method,
+                args
```
Comment (mkordas) [inline src/main/java/com/jcabi/aspects/aj/MethodValidator.java]: @gmandnepr those 3 params could be in one line

**Comment by mkordas** [comment]
@gmandnepr besides that it's good to merge

**Comment by mkordas** [comment]
@gmandnepr good work

**Comment by mkordas** [comment]
@rultor merge

**Comment by mkordas** [comment]
@yegor256 ping

**Comment by yegor256** [comment]
@rultor merge

**Comment by mkordas** [comment]
@rultor deploy

**Comment by yegor256** [comment]
@rultor merge

**Comment by mkordas** [comment]
@yegor256 the above was deployment request so that puzzle is closed faster

**Comment by dmarkov** [comment]
@mkordas I added **30 mins** to your account, many thanks for working with the project (56a74f312550cb0ca200015b)! The completion time here was 142 hours and 15 mins.. extra minutes for review comments (c=15). added +30 to your rating, now it is equal to [+3542](http://www.netbout.com/b/36080?open=rating)

**Comment by dmarkov** [comment]
@rultor deploy pls

**Pull request merged**